### ISSUE TITLE
Port request for Advanced Forum

### ISSUE THREAD

**Issue by djzwerg** [backdrop-ops/contrib#475]
**Name of the module, theme, or layout**
Advanced Forum

**Link to the drupal.org module, theme, or layout**
https://www.drupal.org/project/advanced_forum

**Comment by cptX** [comment]
I would like to have it too the soonest possible. I plan to release a big forum the next month and I was wondering if this module become eventually available if I can install it and use it if I already have users, topics and comments.
At the moment I try to format forum pages and descriptions by myself but if the module could give some enhancements it would be great. The core forum module is too simple and I'm afraid it is not convincing as a forum too much...

**Comment by stpaultim** [comment]
@cptX - I saw your comment here: https://github.com/backdrop-ops/contrib/issues/34#issuecomment-963888900

I think it worth noting that there has been some discussion about improving upon the Drupal 7 forum module, possibly as an alternative to this module. Although, I am sure that on one would object to having this module ported. 

I see that you are aware of the work being done on forum_ng (Forum Next Generation?). 
https://github.com/backdrop-contrib/forum_ng

I can't help but wonder if there is a reason that this module has not been ported yet?

Has anyone tried and found it difficult or problematic?

Is it a particularly difficult module to port?

**Comment by cptX** [comment]
@stpaultim Thanks for replying, I don't think it would be so hard to port as it is mainly based on the existing core forum module. Probably the lack of enough interest paused the process, although at the moment there is no reliable and complete solution in backdrop to impement a good forum. Personally that I need to create a forum I consider this a big problem.

**Comment by richardlampitt** [comment]
I might be working on a migration for this module as part of work for a client in the near future. 

Any pointers / advice would be gratefully recieved.

**Comment by yorkshire-pudding** [comment]
Hi @richardlampitt - how is this port going? I may be able to help if a project I'm looking at goes ahead.

**Comment by richardlampitt** [comment]
> Hi [@richardlampitt](https://github.com/richardlampitt) - how is this port going? I may be able to help if a project I'm looking at goes ahead.

@yorkshire-pudding 

Extremely slowly / on hold for now 😅.

I've got a codebase that I've popped through the `coder-upgrade` module, which I have then gone over manually and provided fallback values for all migrated configuration calls - something that Backdrop's config function doesn't natively support in the same manner that Drupal 7's does. 

Example:
```php
$example = $config->get('advanced_forum_example') ?? [ "fallback" => "array to prevent errors" ];
```

The client that we're migrating for in this instance has elected to skip the advanced forum migration for now, but only after I had started the process. 

Presently at least the codebase I've got here doesn't cause any immediate fatal errors, but I've not had chance to properly test it. 

Not at all averse to tag-teaming this, by the way — but I'm not terribly familiar with the norms of github collaboration; I've only ever needed to work on private repos of <= 3 people before.

**Comment by yorkshire-pudding** [comment]
> ... then gone over manually and provided fallback values for all migrated configuration calls - something that Backdrop's config function doesn't natively support in the same manner that Drupal 7's does.

@richardlampitt - the reason Backdrop doesn't need to do this is that modules can provide a default config file, which D7 didn't do.

The client I'm working with is not bothered about the extra functionality and my looking at it suggests that with all the ctools stuff it could be tricky to un-pick.  Instead I've worked out a migration path for my client to move to [Forum NG](https://github.com/backdrop-contrib/forum_ng) which I think will be easier to customise as it is uses Views.

**Comment by richardlampitt** [comment]
> The client I'm working with is not bothered about the extra functionality and my looking at it suggests that with all the ctools stuff it could be tricky to un-pick. 

This was the main issue that paused our migration for this module. I'd gotten to the point where I was trying to figure out what ctools was doing, how it was doing it, and for what purpose (I have very little Drupal back-end dev experience, so how ctools works and what it does is all new to me).

> Instead I've worked out a migration path for my client to move to [Forum NG](https://github.com/backdrop-contrib/forum_ng) which I think will be easier to customise as it is uses Views.

This would be very helpful indeed!

Ultimately I managed to replicate what our client needed for their site at this phase using a couple of new custom views and a bit of templating for the author panes.
